src/distributions/uniform.rs

```diff
@@ -843,6 +843,26 @@ impl UniformSampler for UniformChar {
     }
 }
 
+/// Note: the `String` is potentially left with excess capacity if the range 
+/// includes non ascii chars; optionally the user may call 
+/// `string.shrink_to_fit()` afterwards.
+#[cfg(feature = "alloc")]
+impl super::DistString for Uniform<char>{
+    fn append_string<R: Rng + ?Sized>(&self, rng: &mut R, string: &mut alloc::string::String, len: usize) {
+        // Getting the hi value to assume the required length to reserve in string.
+        let mut hi = self.0.sampler.low + self.0.sampler.range;
+        if hi >= CHAR_SURROGATE_START {
+            hi += CHAR_SURROGATE_LEN;
+        }
+        // Get the utf8 length of hi to minimize extra space.
+        // SAFETY: hi used to be valid char.
+        // This relies on range constructors which accept char arguments.
+        let max_char_len = unsafe { char::from_u32_unchecked(hi).len_utf8() };
```

Comment:
I don't believe `char::len_utf8` even cares whether `hi` is a valid `char` so this should be safe, but the use of `unsafe` still feels unnecessary. Suggestion:
```
let max_char_len = char::from_u32(hi).map(char::len_utf8).unwrap_or(4);
```

Reply:
I have applied your suggestion.